### DISCUSSION TITLE
New Release v0.6.4

### DISCUSSION THREAD

**Discussion by freakent** [freakent/dbus-mqtt-devices#85] [Announcements]
With the general release of VenusOS v3.20, we also have a new release of dbus-mqtt-devices v0.6.4. 

**I am happy to confirm that Victron's move to FlashMQ from Mosquitto has not impacted this driver and it works just as it did before.** 

This release contains a few bug fixes and some improvements to the samples. Special thanks to @garshaw for providing a fix to a difficult to find bug.

Any comments or feedback please comment down below.

**Comment by gurkc006** [comment]
Good morning freakent!
Good news! I tried the new 0.6.4 this morning with VenusOS 3.20 stable and everything seems to work properly! All my "mqtt"-devices are working as before. Many thanks!
Greetings Christian!

**Comment by ronstekk** [comment]
Good morning!
I also can confirm that everything runs fine with Firmware 3.20.

Thx for the superfast reaction! (maybe faster than flashmq ;-) )

have a nice day!
Cheers,
Ron

**Comment by Thxeverybody** [comment]
Also went back from latest beta to 0.6.4 with release of firmware 3.20.  0.6.4 works fine after executing topic 4 of the Troubleshooting chapter. :-)

**Reply by freakent** [reply]
This one?
4. If you have re-installed more than once, make sure there is only one line in your rc.local for dbus-mqtt-devices.
```
more /data/rc.local
```

**Reply by Thxeverybody** [reply]
Yes, I had still another line referring to 0.7.5

**Reply by freakent** [reply]
I haven't figured out a safe way to do this automatically in the setup script yet. The rc.local file can be used by other services that have been installed. Come to think of it a warning if multiple dbus-mqtt-devices is found is probably better than nothing. I'll add it to the to do list list.

**Reply by Thxeverybody** [reply]
Automatic correction would of course be comfortable, but your advice under troubleshooting and here is already a major help.

**Comment by gharshaw** [comment]
@freakent , I just installed v0.6.4 on Venus v3.30~6 on a Raspberry Pi 3B+ and except for the above mentioned issue with rc.local everything looks good.
Many thanks,
Gene

**Reply by freakent** [reply]
Thanks for proposing the int(...) fix. The function that returns the id already has an int(...) around the return value but for some reason it became a string. Must be one of those Pythonisms.
